fvm/handler/programs_test.go

```diff
@@ -120,6 +128,33 @@ func Test_Programs(t *testing.T) {
 	var contractBView *delta.View = nil
 	var txAView *delta.View = nil
 
+	t.Run("contracts can be updated", func(t *testing.T) {
+		retrievedContractA, err := accounts.GetContract("A", addressA)
+		require.NoError(t, err)
+		require.Empty(t, retrievedContractA)
+
+		// deploy contract A0
+		procContractA0 := fvm.Transaction(contractDeployTx("A", contractA0Code, addressA), 0)
+		err = vm.Run(context, procContractA0, mainView, programs)
+		require.NoError(t, err)
+
+		retrievedContractA, err = accounts.GetContract("A", addressA)
+		require.NoError(t, err)
+
+		require.Equal(t, contractA0Code, string(retrievedContractA))
+
+		// deploy contract A
+		procContractA := fvm.Transaction(contractDeployTx("A", contractACode, addressA), 1)
```

Comment:
you need to call update contract tx like this one 
```
updateContractTx := func(name, code string, address flow.Address) *flow.TransactionBody {
		encoded := hex.EncodeToString([]byte(code))

		return flow.NewTransactionBody().SetScript([]byte(fmt.Sprintf(`transaction {
              prepare(signer: AuthAccount) {
                signer.contracts.update__experimental(name: "%s", code: "%s".decodeHex())
              }
            }`, name, encoded)),
		).AddAuthorizer(address)
	}
```

Reply:
Hmmm, interesting. So now second-deployment is a NOOP? Should it rather error out maybe?